benchmark/shortest_path_benchmark.dart

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2018, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:collection';
+import 'dart:math' show Random;
+
+import 'package:graphs/graphs.dart';
+
+void main() {
+  final _rnd = Random(1);
+  final size = 1000;
+  final graph = HashMap<int, List<int>>();
+
+  for (var i = 0; i < size * 5; i++) {
+    final toList = graph.putIfAbsent(_rnd.nextInt(size), () => List<int>());
+
+    final toValue = _rnd.nextInt(size);
+    if (!toList.contains(toValue)) {
+      toList.add(toValue);
+    }
+  }
+
+  var counts = <int>[];
+
+  final testOutput =
+      shortestPath(0, size - 1, (e) => graph[e] ?? []).toString();
+  print(testOutput);
+  assert(testOutput == '[258, 252, 819, 999]');
```

Comment:
how do you actually know this is correct though? :D

Reply:
this is just a benchmark!